### ISSUE TITLE
hplip requires /etc/hp/hplip.conf

### ISSUE THREAD

**Issue by pceiley** [clearlinux/distribution#516]
*Please omit irrelevant data and trim the bug report template to only those parts that make sense!*

**Describe the bug**
scripts included as part of the hplip package, such as hp-setup rely on /etc/hp/hplip.conf, which is not deployed as part of the package install.

**To Reproduce**
1. run hp-setup
```
$ hp-setup
warning: python-dbus not installed.
error: Unable to locate models.dat file
...
```
2. Create working /etc/hp/hplip.conf file
3. Run hp-setup again
4. The models.dat error disappears

**Expected behavior**
hp-setup should run without errors. Currently it fails due to missing files.

**Screenshots**

**Environment (please complete the following information):**
 - Clear Linux OS version: [`cat /usr/lib/os-release`]
 - Bundles: [e.g. `sudo swupd bundle-list`]
I don't have access to this at present.

**Additional context**
See #383 for more troubleshooting info.

**Comment by ahkok** [comment]
Technically really an upstream issue. We need to patch it to be `stateless`.

**Comment by ahkok** [comment]
`hplip-3.18.6-12`

**Comment by ahkok** [comment]
I've made this stateless - there is a default config file in /usr/share/defaults/etc/hp/ that will be parsed if the version in /etc/hp is missing. The models.dat is handled the same way - it will find a default one. This makes `hp-setup` work for me - at least, enough that I think it works OK.

**Comment by pceiley** [comment]
Thanks, this resolved the models.dat error. I still, however, get python errors:

```
# hp-setup -i 192.168.1.11

HP Linux Imaging and Printing System (ver. 3.18.6)
Printer/Fax Setup Utility ver. 9.0

Copyright (c) 2001-15 HP Development Company, LP
This software comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to distribute it
under certain conditions. See COPYING file for more details.

(Note: Defaults for each question are maked with a '*'. Press <enter> to accept the default.)


Setting up device: hp:/net/OfficeJet_Pro_6970?ip=192.168.1.11


Traceback (most recent call last):
  File "/usr/bin/hp-setup", line 426, in <module>
    ppds = cups.getSystemPPDs()
  File "/usr/share/hplip/prnt/cups.py", line 329, in getSystemPPDs
    path = os.path.join(cups_ppd_path, ppd)
  File "/usr/lib/python3.7/posixpath.py", line 80, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType

```

**Comment by pceiley** [comment]
Thanks. Looking forward to this Lansing to test it out.

**Comment by pceiley** [comment]
Thanks. Looking forward to this landing to test it out.

**Comment by pceiley** [comment]
Thanks very much for your work on this, @ahkok - it's much appreciated!

This issue now appears resolved, but I'm now experiencing #530.
